### DISCUSSION TITLE
Parameters

### DISCUSSION THREAD

**Discussion by zsolt777** [dotnet/WatsonWebserver#65] [Q&A]
Hi @jchristn!

Is it possible to pass parameters to the methods, like:
```
        [StaticRoute(HttpMethod.GET, "/x/Test1")]
        public static async Task Test1(HttpContext ctx, int x)
        {
            await ctx.Response.Send(x.ToString());
        }
```
or using POST to pass a class
```
        [StaticRoute(HttpMethod.POST, "/x/Test2")]
        public static async Task Test2(HttpContext ctx, MyClass myClass)
        {
            await ctx.Response.Send(myClass.Name);
        }
```

Thank you!
Zsolt

**Comment by jchristn** [comment]
Hi @zsolt777, looks like you're referring to ```int x``` and ```MyClass myClass``` as being the request body.  Currently the method has to read the request body and convert/deserialize into the expected form.  

It's likely possible through reflection to determine the type of object expected by the invoked method.  I'd be open to integrating a PR on it if you'd like to take a swing!  Cheers, Joel
